hyperspy/misc/export_dictionary.py

```diff
@@ -0,0 +1,236 @@
+from copy import deepcopy
+try:
+    import dill
+    dill_avail = True
+except ImportError:
+    dill_avail = False
+    import types
+    import marshal
+
+
+def check_that_flags_make_sense(flags):
+# one of: fn, id, sig
+# if fn: no inav, isig
+# if id: no inav, isig
+    def do_error(f1, f2):
+        raise ValueError(
+            'The flags "%s" and "%s" are not compatible' %
+            (f1, f2))
+    if 'fn' in flags:
+        if 'id' in flags:
+            do_error('fn', 'id')
+        if 'sig' in flags:
+            do_error('fn', 'sig')
+        if 'inav' in flags:
+            do_error('fn', 'inav')
+        if 'isig' in flags:
+            do_error('fn', 'isig')
+    if 'id' in flags:
+        # fn done previously
+        if 'sig' in flags:
+            do_error('id', 'sig')
+        if 'inav' in flags:
+            do_error('id', 'inav')
+        if 'isig' in flags:
+            do_error('id', 'isig')
+        if 'init' in flags:
+            do_error('id', 'init')
+    # all sig cases already covered
+
+def parse_flag_string(flags):
+    return flags.replace(' ', '').split(',')
+
+def export_to_dictionary(target, whitelist, dic, picklable=False):
+    """ Exports attributes of target from whitelist.keys() to dictionary dic
+        All values are references only by default.
+        If picklable=True, the functions are copies (hence allows arbitrary closure)
+
+        Parameters
+        ----------
+            target : object
+                must contain the (nested) attributes of the whitelist.keys()
+            whitelist : dictionary
+                A dictionary, keys of which are used as attributes for exporting. Key 'self' is only available
+                with tag 'id', when the id of the target is saved.
+                The values are either None, or a tuple, where:
+                    - the first item a string, which containts flags, separated by commas.
+                    - the second item is None if no 'init' flag is given, otherwise the object required for
+                      the initialization.
+                The flag conventions are as follows:
+                * 'init':
+                    object used for initialization of the target. The object is saved in the tuple in
+                    whitelist
+                * 'fn':
+                    the targeted attribute is a function, and may be pickled (preferably with dill package).
+                    A tuple of (thing, value) will be exported to the dictionary, where thing is None if
+                    function is passed as-is, and bool if dill package is used to pickle the function, and
+                    value is the result.
+                * 'id':
+                    the id of the targeted attribute is exported (e.g. id(target.name) )
+                * 'sig':
+                    The targeted attribute is a signal, and will be converted to a dictionary if
+                    picklable=True
+                * 'inav':
```

Comment:
That's cool, I didn't know that you've implemented this feature in a general way. If you have, again this should go to its own PR and be documented in the User Guide and listed in the highlights.